test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -382,14 +384,16 @@ private void waitData(Map<String, Table<StringKey, IntValue, Metadata>> tables,
     }
 
     private void verifyData(CorfuStore corfuStore, Map<String, Table<StringKey, IntValue, Metadata>> tables,
-                            Map<String, Map<String, Integer>> hashMap) {
+                            Map<String, Map<String, Integer>> hashMap) throws Exception {
         for (String name : hashMap.keySet()) {
             Table<StringKey, IntValue, Metadata> table = tables.get(name);
             Map<String, Integer> map = hashMap.get(name);
 
-            log.debug("Table[" + name + "]: " + table.count() + " keys; Expected "
-                    + map.size() + " keys");
-
+            while (table.count() != map.size()) {
```

Comment:
what if table size never become equal to map size()? test would time out?

Reply:
yes if that happens it indicates an error and tests are supposed to time out. the table size equality check in some tests are aiming for eventual consistency, they do log entry sync asynchronously and then check the equality directly, which is the cause for the flakiness